packages/packagers/js/src/ScopeHoistingPackager.js

```diff
@@ -741,10 +741,28 @@ ${code}
         this.hoistedRequires.set(dep.id, hoisted);
       }
 
-      hoisted.set(
-        resolvedAsset.id,
-        `var $${publicId} = parcelRequire(${JSON.stringify(publicId)});`,
+      let resolvedAssetUsedSymbols = nullthrows(
+        this.bundleGraph.getUsedSymbols(resolvedAsset),
       );
+
+      // Check to see if the exportSymbol is in the resolved asset or the dependency's
+      // set of used symbols. If the exportSymbol is 'default', check that the
+      // default export from the resolved asset is used.
+      // If the exportSymbol is '*', a non-empty set of the resolved asset's used symbols
+      // indicates the export(s) is used.
+      let isExportSymbolUsed =
+        resolvedAssetUsedSymbols.has(exportSymbol) ||
+        nullthrows(this.bundleGraph.getUsedSymbols(dep)).has(exportSymbol) ||
+        (exportSymbol === 'default' &&
+          resolvedAssetUsedSymbols.has(exportSymbol)) ||
```

Comment:
Doesn't the first condition already cover this?

Reply:
the second condition covers the case where a default export is re-exported as some named export:
```
// a.js
import {foo} from './b.js';

// b.js
export {default as foo} from './c.js';

// c.js
export default function(name) { ... }
```
`foo` is in the usedSymbols set of the './b' dep on a.js (whereas c's usedSymbols only contains `default`)

Reply:
You have `resolvedAssetUsedSymbols.has(exportSymbol)` on line 754, which implies that `exportSymbol === 'default' && resolvedAssetUsedSymbols.has(exportSymbol)` is true, right?

Reply:
true! that condition's redundant and has been removed.